### DISCUSSION TITLE
Bad Gateway with traefik and rallly

### DISCUSSION THREAD

**Discussion by fibis** [lukevella/rallly#945] [Q&A] [answered]
We are using rallly together with Traefik as reverse proxy. I have upgraded rallly from version 2 to 3.

We always worked with a "wrong" setup, where we didn't add the scheme to the public base, so ´NEXT_PUBLIC_BASE_URL=example.com` instead of `NEXT_PUBLIC_BASE_URL=https://example.com`. This was working great with version two, but in version 3, we're now getting an `ERR_INVALID_URL`-Error. So I switched to the correct `NEXT_PUBLIC_BASE_URL` and now we are getting a Bad Gateway from Traefik. Directly accessing the application is still possible.

Is there something happening, that is causing this Bad Gateway error, when changing this env?

**Comment by lukevella** [comment]
I've never used traefik so I don't know anything about it but it sounds like it's not forwarding all the request headers. In nginx you would use `proxy_pass` to tell it to forward all headers. Not sure what the equivalent in traefik is but hope that helps.

**Reply by fibis** [reply]
@lukevella that was my first idea. But I only change with `NEXT_PUBLIC_BASE_URL` the behavior of rallly, not of the load balancer. That is the part, that I am not understand.

**Comment by corvec** [comment, marked as answer]
I'm also using Traefik and when updating from an earlier version of v3\*  I also started getting Bad Gateway errors. I reverted to my old version and that solved the issue. I'm not sure if you're experiencing the same issue, but just in case:

Today I noticed https://github.com/lukevella/rallly/issues/947 and tried out [the fix suggested there](https://github.com/lukevella/rallly/issues/947#issuecomment-1845224357) by @scottwallacesh. I added `hostname: 0.0.0.0` and switched to the `lukevella/rallly:3.4.2` image and that solved the Bad Gateway error I'd been seeing. Might be worth a try for you, too.

\* - I'm unsure which version exactly. I pulled it on September 20th from the `lukevella/rallly:3` tag, but the image isn't tagged anymore in dockerhub. The ID was `sha256:0f1dfe32acfd81fd85dfc8c28031be629a6731a4f78280cd11a7d0216f5e3df0`

EDIT: My docker-compose config is [here](https://gist.github.com/corvec/fa95a8e97d730d4c25548576b422630c)

**Reply by fibis** [reply]
@corvec thanks, adding the hostname to the docker-compose file fixed the problem.
